test/client/api.js

```diff
@@ -61,11 +61,20 @@ ClientApi.prototype.doRequest = function (method, url, token, payload, headers)
 
     this.emit('endRequest', options, err, res)
     if (err || body.error || res.statusCode !== 200) {
-      d.reject(err || body)
+      return d.reject(err || body)
     }
-    else {
-      d.resolve(body)
+
+    var allowedOrigin = res.headers['access-control-allow-origin']
+    if (allowedOrigin) {
+      // Requiring config outside this condition causes the local tests to fail
+      // because tokenLifetimes.passwordChangeToken is -1
+      var config = require('../../config')
```

Comment:
I don't know all the intricacies of node's `require` semantics, is this going to have much overhead or other unwanted side-effects if it gets executed multiple times per test across many of our tests?